tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -262,6 +409,136 @@ static void test_tx_cb_multi(void)
 	PASS();
 }
 
+static void test_tx_send_order(void)
+{
+	struct net_buf *buf[CONFIG_BT_MESH_ADV_BUF_COUNT];
+	int err;
+	uint8_t xmit = BT_MESH_TRANSMIT(2, 20);
+
+	static const struct bt_mesh_send_cb send_cb = {
+		.start = send_order_start_cb,
+		.end = send_order_end_cb,
+	};
+
+	bt_init();
+	adv_init();
+	k_sem_init(&observer_sem, 0, 1);
+
+	/* Verify sending order */
+	allocate_all_net_buf(buf, xmit);
+	send_all_net_buf(&buf[0], &send_cb, false);
+
+	/* Wait for no message receive window to end. */
+	err = k_sem_take(&observer_sem, K_SECONDS(10));
+	ASSERT_OK(err, "Didn't call the last end tx cb.");
+	ASSERT_EQUAL(seq_checker, CONFIG_BT_MESH_ADV_BUF_COUNT);
```

Comment:
```suggestion
	ASSERT_OK(k_sem_take(&observer_sem, K_SECONDS(10)), "Didn't call the last end tx cb.");
	ASSERT_EQUAL(seq_checker, CONFIG_BT_MESH_ADV_BUF_COUNT);
```
You could get rid of `err` all together. Up to you. Applies in other tests as well.